.github/workflows/build-image.yaml

```
@@ -0,0 +1,56 @@
+name: Build and Publish CI image
+
+permissions:
+  packages: write
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "images/build/*"
+      - ".github/workflows/build-image.yaml"
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "images/build/*"
+      - ".github/workflows/build-image.yaml"
+
+jobs:
+  build:
+    if: github.repository_owner == 'kcp-dev'
+    name: Build CI Image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      - name: Docker Metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ghcr.io/${{ github.repository_owner }}/infra/build
+          tags: |
+            type=semver,pattern={{version}},value=v1.19.9-1
+
+      - name: Build and Push
+        uses: docker/build-push-action@v4
+        with:
+          context: "{{defaultContext}}:images/build"
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          platforms: linux/amd64
+          cache-from: type=gha,scope=${{ github.ref_name }}-buildx
+          cache-to: type=gha,scope=${{ github.ref_name }}-buildx,mode=max
```

We can pass build args like this.

```suggestion
          cache-to: type=gha,scope=${{ github.ref_name }}-buildx,mode=max
          build-args: GO_VERSION=1.19.1,K8S_VERSION=1.26.3
          
```